### PR TITLE
Update supported Python versions to drop 3.6, include 3.9 and 3.10

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes    

This PR updates supported Python versions based on [current EOL status](https://devguide.python.org/#status-of-python-branches).

- Add variable for Augur version associated with min Python version: https://github.com/nextstrain/augur/commit/3f0974706b35f1ef3b886c1ab86fc1d3a487b77b
- Update CI matrix and minimum version references: https://github.com/nextstrain/augur/commit/65c736fdce7809385d2ffb5771b7490bc3059819
- Bump Augur version associated with min Python version to 14: https://github.com/nextstrain/augur/commit/6451394b0e594082a533de526f4f538573be773a

Note: I didn't update minimum version for biopython, since I'm not sure what we want to support. I'm thinking we bump from 1.67 to 1.73 like https://github.com/nextstrain/augur/commit/3dd51603b0856b9d5c7ee0220ee3a5bca13d42d4 (not in PR), would that be reasonable? We could also try a `matrix` based on [this discussion](https://github.com/nextstrain/augur/pull/796/files/e8fde5c0ef5e6fe0cef022dc87ca113346be8c83#r776533627).

### Related issue(s)
- #793

### CODE REVIEW

**Review by huddlej — APPROVED** [review]
Looks good! We will also need to make a separate PR for the Bioconda recipe to update the minimum Python versions for [host](https://github.com/bioconda/bioconda-recipes/blob/5896748ff0736f45c63c771118ab8d07e2b0eec0/recipes/augur/meta.yaml#L20) and [run](https://github.com/bioconda/bioconda-recipes/blob/5896748ff0736f45c63c771118ab8d07e2b0eec0/recipes/augur/meta.yaml#L24).

I like the idea of using a BioPython matrix (in a separate PR) instead of hardcoding a specific, for the reasons @tsibley lists in the discussion you linked to. The main issue being that we could have passing tests for the old version of BioPython but failing tests for newer versions.

As noted in the `since_augur_version` variable in `setup.py`, we will need a major release for this PR. Since #839 may also introduce a breaking change to the UI for augur tree, we might release these both at the same time in 14.0.0.

**Comment by victorlin** [comment]
@huddlej

> We will also need to make a separate PR for the Bioconda recipe to update the minimum Python versions

At what stage in the release process should this happen? I assume it would be best to somehow link this to the auto-bump PR once we push the new version to pip, but we would have to be diligent to catch it before a maintainer merges. Or do they have a more manual process for major version changes?

**Comment by victorlin** [comment]
Nevermind, I see you answered my question a couple months ago in #793. I'll plan to open a draft PR in bioconda, then link it to the auto-bump PR once that happens. Shouldn't be difficult to catch it before they merge.

**Pull request merged**
